runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/external/ExternalPythonExecutor.java

```diff
@@ -218,34 +251,31 @@ private Serializable processReturnResult(EvaluationResults results) throws JsonP
             case INTEGER:
                 return Integer.valueOf(results.getReturnResult());
             case LIST:
-                return objectMapper.readValue(results.getReturnResult(), new TypeReference<ArrayList<Serializable>>(){});
+                return objectMapper.readValue(results.getReturnResult(), new TypeReference<ArrayList<Serializable>>() {
+                });
             default:
                 return results.getReturnResult();
         }
     }
 
-    private String getResult(String payload, ProcessBuilder processBuilder) throws IOException, InterruptedException {
-        Process process = processBuilder.start();
-        PrintWriter printWriter = new PrintWriter(new OutputStreamWriter(process.getOutputStream(), UTF_8));
-        printWriter.println(payload);
-        printWriter.flush();
-        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
-        String line;
-        StringBuilder returnResult = new StringBuilder();
-        while ((line = reader.readLine()) != null) {
-            returnResult.append(line);
-        }
-        boolean isInTime = process.waitFor(EXECUTION_TIMEOUT, MINUTES);
-        if (!isInTime) {
-            process.destroy();
-            throw new RuntimeException("Execution timed out");
-        } else if (process.exitValue() != 0) {
-            StringWriter writer = new StringWriter();
-            IOUtils.copy(process.getErrorStream(), writer, UTF_8);
-            logger.error(writer.toString());
-            throw new RuntimeException("Execution returned non 0 result");
+    private String getResult(final String payload, final ProcessBuilder processBuilder, final long timeoutPeriod, ExecutorService executorService) throws InterruptedException, ExecutionException {
+        ExternalPythonEvaluationSupplier supplier = new ExternalPythonEvaluationSupplier(processBuilder, payload);
+        long timeoutMillis = timeoutPeriod * 60 * 1000;
+        try {
+            return CompletableFuture.supplyAsync(supplier, executorService).get(timeoutMillis, MILLISECONDS);
+        } catch (TimeoutException e) {
+            try {
+                Process process = supplier.getProcess();
+                if (process != null) {
+                    process.destroy();
+                }
+            } catch (Exception t) {
+                logger.error("Failed to run script. ", t.getCause() != null ? t.getCause() : t);
```

Comment:
should swallow this exception as we need to throw RuntimeException with TimeoutException as cause.